Crash when restoring account

# Bug Report

## Description

When I try to restore an account that has money and communities on testnet, including owner tokens, it crashes.

Here is the seed phrase:
```
awesome fee remove flag cable powder link clog eyebrow noodle window hold
```


## Steps to reproduce

1. Open app
2. "Add existing Status user"
3. "I don't have other device"
4. "Enter a seed phrase"
5. Enter the seed phrase above
6. Press Restore
7. Crash

Not many useful logs:
```
WRN 2024-02-27 13:57:06.363-05:00 Error decoding signal                      topics="signals-manager" tid=1152287 file=signals_manager.nim:49 err="Unknown signal received: mediaserver.started"
2024-02-27T13:57:06-0500 unset [github.com/anacrolix/torrent.(*Client).NewAnacrolixDhtServer.func2:406]: dht server on [::]:43747 (node id 4a4f13ec58087baf8c05cddf9fe8b165cd6dabc9) completed bootstrap ({NumAddrsTried:11 NumResponses:0})
DBG 2024-02-27 13:57:07.351-05:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=1152300 file=threadpool.nim:54 messageType=TimerTaskArg:ObjectType threadid=1152300 task="{\"$type\":\"TimerTaskArg:ObjectType\",\"timeoutInMilliseconds\":1000,\"reason\":\"\",\"vptr\":94524747352432,\"slot\":\"onTimeout\",\"tptr\":94524712399600}"
2024-02-27T13:57:07.660-0500	INFO	communities/manager.go:371	initializing history archive tasks
DBG 2024-02-27 13:57:08.353-05:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=1152302 file=threadpool.nim:54 messageType=TimerTaskArg:ObjectType threadid=1152302 task="{\"$type\":\"TimerTaskArg:ObjectType\",\"timeoutInMilliseconds\":1000,\"reason\":\"\",\"vptr\":94524747352432,\"slot\":\"onTimeout\",\"tptr\":94524712399600}"
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
make: *** [Makefile:815: run-linux] Segmentation fault (core dumped)
```

#### Expected behavior

The restoration works or at least doesn't crash

#### Actual behavior

Crashes almost straight away.

### Additional Information

- Status desktop version: master `6de7032d65386b8d59168ef1908a3866aa3b854d
- Operating System: Ubuntu 20

--- a/protocol/messenger_backup_handler.go
+++ b/protocol/messenger_backup_handler.go
@@ -181,7 +181,9 @@ func (m *Messenger) handleBackedUpProfile(message *protobuf.BackedUpProfile, bac
 	if err != nil {
 		return err
 	}
-	response.SetProfileShowcasePreferences(profileShowcasePreferences)
+	if profileShowcasePreferences != nil {
+		response.SetProfileShowcasePreferences(profileShowcasePreferences)
+	}
 
 	var ensUsernameDetails []*ensservice.UsernameDetail
 	for _, d := range message.EnsUsernameDetails {

--- a/protocol/messenger_profile_showcase.go
+++ b/protocol/messenger_profile_showcase.go
@@ -669,6 +669,9 @@ func (m *Messenger) DeleteProfileShowcaseCommunity(community *communities.Commun
 }
 
 func (m *Messenger) saveProfileShowcasePreferencesProto(p *protobuf.SyncProfileShowcasePreferences, shouldSync bool) (*identity.ProfileShowcasePreferences, error) {
+	if p == nil {
+		return nil, nil
+	}
 	preferences := FromProfileShowcasePreferencesProto(p)
 	return preferences, m.setProfileShowcasePreferences(preferences, shouldSync)
 }

